src/main/java/com/joeyfrazee/nifi/reporting/ElasticsearchProvenanceReporter.java

```diff
@@ -257,4 +349,32 @@ private ElasticsearchClient getElasticsearchClient(RestClient restClient) {
                 new RestClientTransport(restClient, new JacksonJsonpMapper());
         return new ElasticsearchClient(transport);
     }
+
+    /**
+     * Return a subset of the given map that either includes the given fields (and no other fields),
+     * or excludes them.
+     *
+     * <p>All whitespace within the given fields is ignored.
+     *
+     * @param map The map.
+     * @param fieldsToFilterString The fields to filter for, as a comma-separated list.
+     * @param include Whether to include or exclude the given fields.
+     * @return The filtered map.
+     */
+    private ImmutableMap<String, Object> getFilteredMap(
+            Map<String, Object> map, String fieldsToFilterString, boolean include) {
```

Comment:
I'd get rid of the `include` and just do the stateful checking directly in here:

```
         // Filter event fields based on inclusion/exclusion list.
        final String inclusionListString = context.getProperty(ELASTICSEARCH_INCLUSION_LIST).getValue();
        final String exclusionListString = context.getProperty(ELASTICSEARCH_EXCLUSION_LIST).getValue();

        if (!Strings.isNullOrEmpty(inclusionListString) {
            filteredEvent = getFilteredMap(event, inclusionListString, true);
        } else if (!Strings.isNullOrEmpty(exclusionListString)) {
            filteredEvent = getFilteredMap(event, exclusionListString, false);
        }
        else {
            // No filtering required.
            return event;
        }
```


Can probably rename the method to `filterEventFields`, then in the earlier code you can just go:
```
        Map<String, Object> filteredEvent = filterEventFields(event);
```